rust/flatbuffers/src/endian_scalar.rs

```diff
@@ -80,92 +87,52 @@ impl_endian_scalar_stdlib_le_conversion!(i32);
 impl_endian_scalar_stdlib_le_conversion!(i64);
 
 impl EndianScalar for f32 {
+    type Scalar = u32;
     /// Convert f32 from host endian-ness to little-endian.
     #[inline]
-    fn to_little_endian(self) -> Self {
-        #[cfg(target_endian = "little")]
-        {
-            self
-        }
-        #[cfg(not(target_endian = "little"))]
-        {
-            byte_swap_f32(self)
-        }
+    fn to_little_endian(self) -> u32 {
+        self.to_bits().to_le()
```

Comment:
This is consistent with the previous behaviour, and will reverse the bytes on big endian, but I'm not entirely sure it is correct. I'm not sure endianess influences floating point numbers?

Reply:
Perhaps leave a comment with this quote and link:

> Floats and Ints have the same endianness on all supported platforms.

https://doc.rust-lang.org/std/primitive.f32.html#method.from_bits